src/pizzaOrder/Pizza.java

```diff
@@ -0,0 +1,124 @@
+package pizzaOrder;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Pizza {
+
+    private String name;
+    private Type type;
+    private List<Ingredient> ingredients;
+    private int quantity;
+
+    public Pizza(String name, Type type, int quantity) {
+        this.checkPizzaName(name);
+        this.type = type;
+        this.quantity = quantity;
+    }
+
+    public Pizza(String name, Type type, List<Ingredient> ingredients, int quantity) {
+        this.checkPizzaName(name);
+        this.type = type;
+        this.ingredients = ingredients;
+        this.quantity = quantity;
+    }
+
+    public void addIngredients(String ingredientName) {
+        if (ingredients == null) {
+            ingredients = new ArrayList<>();
+        }
+        if (ingredients.size() == 7) {
+            System.out.println("You can't add any more ingredients");
+            return;
+        }
+        for (Ingredient ingredient1 : ingredients) {
+            if (ingredient1.getName().equals(ingredientName)) {
+                throw new IllegalArgumentException("You have already added that ingredient");
+            }
+        }
+        for (Ingredient ingredient : Ingredient.values()) {
+            if (ingredient.getName().equals(ingredientName)) {
+                ingredients.add(ingredient);
+            }
+        }
+    }
+
+    private void checkPizzaName(String name) {
```

Comment:
Now the second part, here you are violating the Single Responsibility principle, please create Separate class for this logic, like PizzaValidator and validate pizza name in that

Reply:
The method has been changed